x-pack/solutions/security/plugins/security_solution/docs/adrs/threat_hunting/0002_timeline_ui_batching_strategy.md

```diff
@@ -0,0 +1,212 @@
+10 pages with each page consisting of 50 documents  
+20 pages with each page consisting of 25 documents
+5 pages with each page consisting of 100 documents.
+
+Users will be able to fetch a new batch of documents when they reach the last page.
+
+### Batch Parameters
+Below are the parameters that affect a batch. Any changes to these parameters should trigger a re-fetch of all existing batches or the initial batch depending on the situation.
+
+- **dataViewId**: Currently selected dataView Id in Timeline.
+- **Timerange**: 
+  - **startDate**
+  - **endDate**
+  - **timeRangeKind**
+- **eqlOptions**
+- **indexNames**: List of indices in Timeline.
+- **id**: Timeline Id
+- **fields**:  List of requested fields
+- **filterQuery**:  combined query consisting: 
+  * **KQL/Lucene Query**
+  * **Filters**
+  * **Data Providers**
+- **runtimeMappings**: List of runtime mappings that need to be fetched
+- **language**: KQL | Lucene
+- **limit**: Sample Size or Batch Size
+- **sort**
+
+## Batch fetch/re-fetch Strategy
+Batch fetch/re-fetch strategy depends on the scenario or the action the user is taking. This section will lay out all the scenarios in which a batch should be re-fetched and its corresponding strategy.
+
+### New Batch Requested
+When a user requests a new batch after clicking on `Load more` as shown below, a new batch will be fetched. For example, if user is on Batch 0 and clicks on Load more, a new batch will be fetched with below strategy :
+
+![](./images/load_more_timeline.png)
+
+#### **Expected fetch/re-fetch logic**
+
+```js
+{
+  /* Fetch Batch 1 */
+  activePage: 1, /* current batch */
+  querySize: 500 /* batch/sample size */
+}
+```
+
+Consequently, if a user requests a new batch again, Batch 2 will be fetched with below strategy.
+
+```js
+{
+	/* Fetch Batch 2 */
+	activePage: 2,
+	querySize: 500 /* batch/sample size */
+}
+```
+#### **Expected UI behavior**
+
+- When a new batch is fetched, the Datagrid should have new pages appended to it.
+ 
+### Refresh Button
+
+When a user clicks on the Refresh button in the query tab, it should explicitly re-fetch 0th batch again irrespective of the fact if any batch parameters have been changed or not.
+
+This also applied to the refetch logic triggered by :
+- operations such as change in the Alert's status.
+- triggering of global query.
+- Clicking Refresh button in the query tab.
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+#### Considerations
+- This logic might lead to degrdations in UX in scenario given below:
+  - when user is on let's say 3rd batch and they change the status of an Alert (which results in refetch). 
+  - This will reset the table page to go back to 1st since we re-fetch 0th batch as mentioned above. 
+  - Team decided that it is rare scenario since many users tend to do these kind of operations on the first page of the Datagrid.
```

Comment:
This section is very confusing to me. It seems that we're listing 3 different bullet points for 3 different scenarios but when reading each bullet point they seem to talk about the same situation?

Reply:
I is actually one scenario. I have reworded and reorganized it [here](https://github.com/elastic/kibana/commit/ba22b4eb28d01c298e12e3c07a988755c827e960#diff-45d1a25f892080a8e562b9763c9124ec07b1adb7d132dc454da4e0455e8637a2R92-R93). Please see if it is better.